analyzer/src/main/kotlin/managers/Cargo.kt

```diff
@@ -0,0 +1,279 @@
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.analyzer.managers
+
+import ch.frankel.slf4k.*
+
+import com.fasterxml.jackson.databind.JsonNode
+
+import com.here.ort.analyzer.AbstractPackageManagerFactory
+import com.here.ort.analyzer.PackageManager
+import com.here.ort.downloader.VersionControlSystem
+import com.here.ort.model.Identifier
+import com.here.ort.model.Package
+import com.here.ort.model.PackageLinkage
+import com.here.ort.model.PackageReference
+import com.here.ort.model.Project
+import com.here.ort.model.ProjectAnalyzerResult
+import com.here.ort.model.RemoteArtifact
+import com.here.ort.model.Hash
+import com.here.ort.model.Scope
+import com.here.ort.model.VcsInfo
+import com.here.ort.model.VcsType
+import com.here.ort.model.config.AnalyzerConfiguration
+import com.here.ort.model.config.RepositoryConfiguration
+import com.here.ort.model.jsonMapper
+import com.here.ort.utils.CommandLineTool
+import com.here.ort.utils.log
+import com.here.ort.utils.textValueOrEmpty
+
+import com.moandjiezana.toml.Toml
+
+import com.vdurmont.semver4j.Requirement
+
+import java.io.File
+import java.util.SortedSet
+
+class Cargo(
+    name: String,
+    analyzerRoot: File,
+    analyzerConfig: AnalyzerConfiguration,
+    repoConfig: RepositoryConfiguration
+) : PackageManager(name, analyzerRoot, analyzerConfig, repoConfig), CommandLineTool {
+    class Factory : AbstractPackageManagerFactory<Cargo>("Cargo") {
+        override val globsForDefinitionFiles = listOf("Cargo.toml")
+
+        override fun create(
+            analyzerRoot: File,
+            analyzerConfig: AnalyzerConfiguration,
+            repoConfig: RepositoryConfiguration
+        ) = Cargo(managerName, analyzerRoot, analyzerConfig, repoConfig)
+    }
+
+    companion object {
+        private const val REQUIRED_CARGO_VERSION = "1.0.0"
+        private const val SCOPE_NAME_DEPENDENCIES = "dependencies"
+        private const val SCOPE_NAME_DEV_DEPENDENCIES = "devDependencies"
+        private val pathDependencyRegex = Regex("""^.*\(path\+file://(.*)\)$""")
+    }
+
+    override fun command(workingDir: File?) = "cargo"
+
+    override fun getVersionRequirement(): Requirement = Requirement.buildStrict(REQUIRED_CARGO_VERSION)
+
+    private fun runMetadata(workingDir: File): String = run(workingDir, "metadata", "--format-version=1").stdout
+
+    private fun extractCargoId(node: JsonNode) = node["id"].textValueOrEmpty()
+
+    private fun extractPackageId(node: JsonNode) =
+        Identifier(
+            type = "Cargo",
+            namespace = "",
+            name = node["name"].textValueOrEmpty(),
+            version = node["version"].textValueOrEmpty()
+        )
+
+    private fun extractRepositoryUrl(node: JsonNode) = node["repository"]?.textValue()
+
+    private fun extractVcsInfo(node: JsonNode): VcsInfo {
+        val url = extractRepositoryUrl(node)
+        val type = url?.let { VcsType.GIT } ?: VcsType.NONE // For now cargo supports only git.
```

Comment:
Any particular reason why you're not using `VersionControlSystem.splitUrl()` here?

Reply:
Ah, much better. Thanks.